File: app/Http/Controllers/SpecialistPatternsJava/Creational/Prototype/PrototypeJava/src/Point.java
public class Point implements Cloneable {
    int x, y;

    // Конструктор класса Point, принимающий координаты x и y.
    public Point(int x, int y) {
        this.x = x;
        this.y = y;
    }

    // Конструктор класса Point, создающий копию существующей точки Point.
    public Point(Point p) {
        this(p.x, p.y);
    }

    // Метод clone() для создания клона (копии) текущего объекта Point.
    @Override
    public Point clone() {
        return new Point(this);
    }

    // Переопределенный метод toString() для вывода информации о Point в виде строки.
    @Override
    public String toString() {
        return String.format("Point (%d,%d)", x, y);
    }
}

File: app/Http/Controllers/SpecialistPatterns/Prototype/Point.php
<?php

namespace App\Http\Controllers\SpecialistPatterns\Prototype;

// Класс "Точка" представляет собой объект точки с координатами x и y.
class Point implements Prototype {
    protected $x;
    protected $y;

    // Конструктор класса, инициализирует координаты x и y.
    public function __construct($x, $y) {
        $this->x = $x;
        $this->y = $y;
    }

    // Метод clone() создает и возвращает клон (копию) объекта точки.
    public function clone() {
        return new Point($this->x, $this->y);
    }

    // Переопределение метода __toString() для получения строкового представления объекта.
    public function __toString() {
        return "Point ({$this->x}, {$this->y})";
    }
}
